t/mojo/promise.t

```diff
@@ -368,6 +368,47 @@ subtest 'Settle with promise' => sub {
   $promise->catch(sub { });
 };
 
+subtest 'Settle with self' => sub {
+  my $promise = Mojo::Promise->new;
+  my (@results, @errors);
+  $promise->resolve($promise)->then(sub { @results = @_ }, sub { @errors = @_ });
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  cmp_ok scalar(@errors), '==', 1, 'correct number of errors';
+  like $errors[0], qr/^Chaining cycle detected for promise \S/, 'correct error';
+
+  $promise = Mojo::Promise->new;
+  (@results, @errors) = ();
+  $promise->reject($promise)->then(sub { @results = @_ }, sub { @errors = @_; () });
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  is_deeply \@errors, [$promise], 'correct errors';
+};
+
+subtest 'Settle with unsettled promise' => sub {
+  my $promise  = Mojo::Promise->new;
+  my $promise2 = Mojo::Promise->new;
+  my @results;
+  $promise->then(sub { @results = @_ });
+  $promise->resolve($promise2);
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  $promise->resolve(123);
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  $promise2->resolve(1_000);
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [1_000], 'promise resolved';
+
+  $promise = Mojo::Promise->new;
+  (@results, my @errors) = ();
+  $promise->then(sub { @results = @_ }, sub { @errors = @_ });
+  $promise->reject($promise2);
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  is_deeply \@errors, [$promise2], 'correct errors';
+};
```

Comment:
This block of tests seems completely unrelated to the PR description.

Reply:
Indeed, this block was related to the second issue this PR solves, which was added later. (For history reasons I thought I shouldn't have modified any of the past comments, including the title & description). Now the PR's description has been updated to address that.

Reply:
PRs trying to resolve multiple issues at once pretty much never get accepted.